transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/AntibodyTransformations.scala

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.monster.encode.transformation
+
+import java.time.OffsetDateTime
+import org.broadinstitute.monster.encode.jadeschema.table.Antibody
+import org.slf4j.LoggerFactory
+import upack.Msg
+
+object AntibodyTransformations {
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  val EncodeIdPattern = "/[^/]+/([^/]+)/".r
+
+  def transformAntibody(antibodyInput: Msg): Antibody = {
+    import org.broadinstitute.monster.common.msg.MsgOps
+
+    val id = antibodyInput.read[String]("accession")
+    val targets = antibodyInput
+      .read[Array[String]]("targets")
+
+    // Use regular expressions to remove everything but the actual target name, which should be the same across
+    // all targets in the list. Remove "synthetic_tag" type targets.
+    val mappedTargets = targets.collect {
+      case EncodeIdPattern(target) => target
+    } // filter out non-matches
+    .flatMap { target =>
+      val (front, back) = target.splitAt(target.lastIndexOf('-'))
+      if (back == "-synthetic_tag") None
+      else Some(front)
+    }
+    val firstMappedTarget = mappedTargets.headOption
+
+    // check that all other target values match the first target value
+    if (!mappedTargets.forall(target => target == firstMappedTarget)) {
```

Comment:
I don't think this equality check will work as-is. `target` has type `String`, but `firstMappedTarget` has type `Option[String]`.

You can simplify the whole thing by rearranging slightly. Add a `.toSet[String]` after the `flatMap`, at the end of the `mappedTargets` definition. The Set builder will de-duplicate values. Then you can:
1. Check if size > 1, and log the warning if so.
2. Take the `headOption` of the set for use in the output model